FOSSGIS/2017/workshop_notes.md

```diff
@@ -0,0 +1,196 @@
+# Preparation step
+
+## Downloading OSM data
+
+
+http://download.geofabrik.de/europe/germany/bayern/niederbayern.html
+
+```sh
+mkdir -p ~/data && cd ~/data
+wget http://download.geofabrik.de/europe/germany/bayern/niederbayern-latest.osm.pbf
+```
+
+*** Setting up the osrm-backend Docker image
+
+https://hub.docker.com/r/osrm/osrm-backend-ci/
+
+```sh
+docker pull osrm/osrm-backend-ci
+```
+
+In the docker image the current work directory will be mounted to /data
+
+```sh
+docker run -v $(pwd):/data osrm/osrm-backend-ci ls /data
+```
+
+
+# OSRM toolchain
+
+
+1. Extraction step converts OSM data using a Lua profile file into a set of OSRM files
+
+```sh
+docker run -v $(pwd):/data osrm/osrm-backend-ci osrm-extract -p /opt/car.lua /data/niederbayern-latest.osm.pbf
+```
+
+2. Contraction step augments the base graph with Contraction Hierarchies
+
+```sh
+docker run -v $(pwd):/data osrm/osrm-backend-ci osrm-contract /data/niederbayern-latest.osrm
+```
+
+3. Run an OSRM server with pre-processed OSRM files
+
+```sh
+docker run -v $(pwd):/data -p 5000:5000 osrm/osrm-backend-ci osrm-routed /data/niederbayern-latest.osrm
+```
+
+Now you can a routing query
+```sh
+curl "http://localhost:5000/route/v1/driving/13.449374,48.566477;13.460645,48.574014" | jq .
+```
+
+```json
+{
+  "code": "Ok",
+  "routes": [
+    {
+      "geometry": "oslgHuyaqAgBEyBqGqAkHeDyGcCmG[yAmAsJmAgEsDhEiBcEyCkDeDnIe@\\]@AqA]gC`BoFmBeBZoA",
+      "legs": [
+        {
+          "steps": [],
+          "distance": 1640,
+          "duration": 273.7,
+          "summary": "",
+          "weight": 273.7
+        }
+      ],
+      "distance": 1640,
+      "duration": 273.7,
+      "weight_name": "routability",
+      "weight": 273.7
+    }
+  ],
+  "waypoints": ...
+}
+```
+
+# OSRM frontend
+
+OSRM frontend https://github.com/Project-OSRM/osrm-frontend
+can be used to visualize route requests.
+
+## Clone repository
+```sh
+mkdir ~/osrm-frontend && cd osrm-frontend
+git clone https://github.com/Project-OSRM/osrm-frontend
+```
+
+
+## Change routing requests to the local OSRM server
+
+In file `~/osrm-frontend/src/leaflet_options.js`
+```
+L31:  services: [{
+L32:    label: 'Car (fastest)',
+L33:    path: 'https://router.project-osrm.org/route/v1'
+L34:  }],
+```
+
+`path` should be changed to 'http://localhost:5000/route/v1'
+
+## Change debug tiles requests to the local OSRM server (optional)
+
+In file `~/osrm-frontend/debug/index.html`
+```
+L126:        "osrm": {
+L127:            "type": "vector",
+L128:            "tiles" : ["http://router.project-osrm.org/tile/v1/car/tile({x},{y},{z}).mvt"]
+L129:        }
+```
+`"tiles"` should be changed to `http://localhost:5000/tile/v1/car/tile({x},{y},{z}).mvt`
```

Comment:
Just added these to https://github.com/Project-OSRM/osrm-frontend to have it upstream :+1: